nx/lib/nx.ex

```diff
@@ -11516,32 +11516,35 @@ defmodule Nx do
   """
   @doc type: :ndim
   def sort(tensor, opts \\ []) do
-    opts = keyword!(opts, axis: 0, direction: :asc)
+    opts =
+      keyword!(opts, axis: 0, direction: :asc)
+      |> vectorized_offset_axis_option(tensor)
 
-    direction =
-      case opts[:direction] do
-        :asc ->
-          :asc
+    apply_vectorized([tensor], fn tensor ->
```

Comment:
Another option is to receive the offset as argument:

    apply_vectorized_axes([tensor], fn tensor, offset ->

    end)

You only pass the offset if arity is 2. Then you add the offset once you compute the axis. It should be less work than manipulating options.

Reply:
I didn't really see a reason to change the generic function to accommodate 2 functions. Perhaps if more functions appear in the future we can do this change?

Reply:
I still don't see how this avoids manipulating options inside the function

Reply:
Maybe we should have `apply_vectorized(inputs, opts, function)` that fixes some options like `:padding_config`, `:axes` and `:axis` automatically and passes it down to the arity-2 function